cosign/README.md

```diff
@@ -0,0 +1,125 @@
+|-----------------------------------|------------------------------------------------------------------------|-----------------------------------------------------------------------------|
+| **Keyless Verification**          | Verifyies signatures from Rekor, trust material (signed certificate timestamp) from the CT log, and certificates that chain up to Fulcio. | CI/CD pipelines or automated workflows where no private key management is needed. |
+| **Key-Based Verification**        | Verifies signature using a known **public key**.                        | Environments where signatures are verified with a known public key. |
+| **Timestamp Verification**        | Verifies the **timestamp** of the signature to prevent time-based attacks. Not standalone verification. | Long-term signature validity checks, ensuring signatures are valid at a specific point in time. |
+| **Rekor Transparency Log (TLog)** | Verifies inclusion of the signature in the **Rekor Transparency Log** for audibility. Not standalone verification. | Auditing and compliance to ensure signatures are publicly recorded in an immutable log. |
+
+### Cosign Verifier Input
+
+1. The Artifact to Verify
+    - The artifact in OCI-compliant registries.
+
+2. Verification Materials
+    - The signature(s) of the artifact[4]: The cryptographic signature(s) created when the artifact was signed.
+    - Verification options: i.e. whether to expect signed certificate timestamps (SCTs), TLog entries, or signed timestamps and expected identity and digest to verify.
+
+3. Verification Key or Certificate
+    - Keyless Verification: Retrieves an X.509[5] certificate on the signature and verify against Fulcio root trust.
+    - Key-Based Verification: User provides a public key, certificate or certificate chain.
+
+### Cosign Verifier Output
+
+The output format is defined by `ratify-go`, illustrates if the signature is valid and all criteria are met. It includes detailed information such as:
+
+- verification_success: Indicates if the verification was successful.
+- verifier reference:  The verifier that generated the result.
+- artifact_digest: The cryptographic hash of the artifact.
+- signer_identity: The identity associated with the signing certificate (if keyless signing was used).
+- signing_cert: Metadata about the Fulcio-issued certificate (if applicable).
+- rekor_entry: Confirms whether the signature exists in the transparency log (for keyless verification).
+- timestamp_verified: Ensures that the signature has a valid timestamp (if timestamping was used).
+
+## References
+
+[1] [Zachary Newman, John Speed Meyers, and Santiago Torres-Arias. 2022. Sigstore: software signing for everybody. In Proceedings of the 2022 ACM SIGSAC Conference on Computer and Communications Security. 2353–2367](https://doi.org/10.1145/3548606.3560596)
```

Comment:
If you don't know how to create bibliography item entry, you may just do the ACM ref
- Zachary Newman, John Speed Meyers, and Santiago Torres-Arias. 2022. Sigstore: Software Signing for Everybody. In Proceedings of the 2022 ACM SIGSAC Conference on Computer and Communications Security (CCS '22). Association for Computing Machinery, New York, NY, USA, 2353–2367. https://doi.org/10.1145/3548606.3560596

Reply:
Updated.